### PR TITLE
Jolly's Homework 04

### PR DESCRIPTION
**Pull request by jollylst**
Jolly's Homework 04, thanks!

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,18 +11,33 @@
...
+	count = 0
+	value = ''
+	while count < 5 and value != x:
+		try:
+			value = int(input('Guess what the number is:\n'))
+			if value == x:
+				print('Your guess is right!')
+			elif value > x:
+				print('Your guess is too high, please try again!')
+			elif value < x:
```
Comment (danielsgriffin) [inline D04_ex00.py]: Should this be an elif?

File: HW04_ch07_ex02.py [hunk 2]
```diff
@@ -21,15 +21,27 @@
...
+	while True:
+		try:
+			equation_input = input('Please enter your equation for evaluation:\n')
+			if equation_input == 'done':
+				return result
+			result = eval(equation_input)
+			print(result)
+		except:
+			print('Invalid input, please try again...')
+			continue
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: Is this continue required?

File: HW04_ch08_ex00.py [hunk 3]
```diff
@@ -17,17 +17,25 @@
...
 # Imports
-
+import math
 
 # Body
+def count(word, letter):
+	count = 0
+	for i in word:
+		if i == letter:
+			count = count + 1
```
Comment (danielsgriffin) [inline HW04_ch08_ex00.py]: count += 1

File: HW04_ch08_ex05.py [hunk 4]
```diff
@@ -2,3 +2,35 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
 # function.
+
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: very nice docstrings!

File: HW04_ch08_ex05.py [hunk 5]
```diff
@@ -2,3 +2,35 @@
...
+	# return: single-letter string
+	if letter.isupper():
+		start = ord('A')
+	elif letter.islower():
+		start = ord('a')
+	else:
+		return letter
+
+	c = ord(letter) - start
+	i = (c + n) % 26 + start
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Good job on this whole problem. 
the % 26 is the sort of line that you might want to explain w/ an inline-comment

**Comment by danielsgriffin** [comment]
Good job.

File: HW04_string2.py [hunk 6]
```diff
@@ -30,8 +35,11 @@ def verbing(s):
 # So 'This dinner is not that bad!' yields:
 # This dinner is good!
 def not_bad(s):
-    # +++your code here+++
-    return
+    s_not = s.find('not')
+    s_bad = s.find('bad')
+    if s_bad > s_not:
```
Comment (priyankapatki) [inline HW04_string2.py]: It might fail with a string that has "bad but no "not". eg: "This is not as bad".

File: HW04_ch08_ex05.py [hunk 7]
```diff
@@ -2,3 +2,35 @@
...
+	# n: integer
+	# return: single-letter string
+	if letter.isupper():
+		start = ord('A')
+	elif letter.islower():
+		start = ord('a')
+	else:
+		return letter
+
+	c = ord(letter) - start
```
Comment (OhRobin) [inline HW04_ch08_ex05.py]: c is the new character? Descriptive names would help

File: HW04_string2.py [hunk 8]
```diff
@@ -42,8 +50,24 @@ def not_bad(s):
...
+        a_back = a[(a_len//2 + 1):]
+
+    if b_len % 2 == 0:
+        b_front = b[:(b_len//2)]
+        b_back = b[(b_len//2):]
+    else:
+        b_front = b[:((b_len//2)+1)]
+        b_back = b[(b_len//2 + 1):]
+
+    return a_front + b_front + a_back + b_back
```
Comment (OhRobin) [inline HW04_string2.py]: Good code, but your floor mod (//) takes care of both cases I believe. you don't need the +1 else statements

**Pull request closed without merging**